docs(admin): document external storage change detection and limitations (#14937)
nextcloud/documentation: admin_manual/configuration_files/external_storage_configuration_gui.rst

* Encryption
* Previews
* Enable Sharing
* **Filesystem check frequency** — controls how the server rescans an external
  storage path when a WebDAV ``PROPFIND`` request arrives for it.

>   **Once per direct access** rescans on each ``PROPFIND`` request;

Reviewer: To prevent any expectation that the whole storage will be scanned.

```suggestion
  **Once per direct access** makes a shallow rescans on each ``PROPFIND`` request;
```

Author: Done as well!

>   **Once per direct access** rescans on the first ``PROPFIND`` per PHP request;

Reviewer: Is it possible to have multiple `PROPFIND` per PHP request?

Author: Simultaneously?
I don't think so, no 🤔

Reviewer: Then the sentence is false

.. figure:: external_storage/images/user_mounts.png
   :alt: Checkboxes to allow users to mount external storage services.

.. _external_storage_change_detection_label:

Detecting changes made outside Nextcloud
-----------------------------------------

When files are added, modified, or deleted on an external storage **directly**
(without going through Nextcloud), Nextcloud will not know about those changes
immediately. Nextcloud maintains an internal index (the *filecache*) that is only
updated when Nextcloud itself performs a scan. Until that scan happens, the
filecache is stale and the changes are invisible to Nextcloud, desktop clients,
and mobile apps.

How the "Filesystem check frequency" option works
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The **Filesystem check frequency** mount option (see
:ref:`external_storage_mount_options_label`) controls what happens when a WebDAV
``PROPFIND`` request reaches the Nextcloud server for a path inside the external
storage:

* **Once per direct access** — the server rescans that directory level on the first

>   ``PROPFIND`` per PHP request, then updates the filecache for what it finds there.

Reviewer: Not sure whether the initial version made sense.
```suggestion
* **Once per direct access** — during ``PROPFIND`` requests made to a given directory, 
  Nextcloud rescans that directory level then updates the filecache with what it finds there.
```

* **Never** — the server skips the rescan entirely and always returns the cached

>   state.

Reviewer: Just to remove the suggestion that the filecache table is a cache.
```suggestion
* **Never** — the server never rescans the external storage during ``PROPFIND`` requests
  and always returns the content stored in its internal index.
```

This option is *only* about the server's behaviour during a ``PROPFIND``. It does
not trigger any background polling and does not change how desktop clients or
mobile apps behave.

Because this rescan is driven by user actions (opening a folder in the web UI,
browsing in the mobile app, or a client issuing a ``PROPFIND``), change detection
remains random and unreliable for deep folder hierarchies.

Why desktop clients miss deep changes
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The desktop sync client only issues a ``PROPFIND`` on the root of the sync folder
on every sync cycle because scanning the entire tree would be too expensive. It
relies on **etag propagation**: when a file changes, the etag of every parent
folder up to the root must also change, giving the client a trail to follow down
to the changed file.

When a change is made outside Nextcloud, no etag is updated. The client sees no
trail from the root and never follows it down to the changed file — even if the
file itself is eventually rescanned at a deeper level, the propagation up to the

> root must still happen before the client can discover it.

Reviewer: > even if the file itself is eventually rescanned at a deeper level, the propagation up to the
root must still happen before the client can discover it.

The propagation happens immediately during the scan, so no need to phrase it as if it is a long process. I would remove the sentence altogether.

```suggestion
trail from the root and never follows it down to the changed file.
```

How to make external changes reliably detectable
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

There are two approaches, depending on the storage type:

**A) SMB/CIFS storages — use update notifications**

The ``files_external:notify`` command listens for change events sent by the SMB
server itself. When a change event arrives, Nextcloud rescans the affected path
and propagates the etag changes up to the root, so desktop clients pick up the
change on their next sync cycle.

.. code-block:: console

   occ files_external:notify <mount_id>

See :doc:`external_storage/smb` for setup details, including authentication
requirements and how to reduce sync delay.

**B) All other storages — periodic rescan via cron**

For storage types that do not support push notifications, set up a cron job that
rescans the storage periodically:

.. code-block:: console

>    sudo -E -u www-data php occ files:scan --path="/user/files/MountName"

Reviewer: We have a `files_external:scan <mount_id>` command nowadays, maybe we should recommend that one.

Or to rescan all files for all users (including all external storages):

.. code-block:: console

>    sudo -E -u www-data php occ files:scan --all

Reviewer: I don't think we should recommend that to admins. This will unnecessarily consume resources.

See :doc:`../occ_command` for the full ``files:scan`` reference. A typical
interval is every 15 minutes; adjust to balance freshness against server load.

.. note::
   On Nextcloud AIO the equivalent command is:

   .. code-block:: console

      sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ files:scan --all

We also recommend configuring the background job as **Webcron** or **Cron**
(see :doc:`../configuration_server/background_jobs_configuration`) so that
Nextcloud's regular housekeeping runs reliably alongside any custom scan cron job.

Limitation: renames are not detected reliably
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> When using periodic rescanning (option B above), the scanner cannot reliably

Reviewer: Is it limited to periodic scans? Isn't it also an issue with the "Once per direct access" change detection setting?

There are two approaches, depending on the storage type:

**A) SMB/CIFS storages — use update notifications**

The ``files_external:notify`` command listens for change events sent by the SMB
server itself. When a change event arrives, Nextcloud rescans the affected path
and propagates the etag changes up to the root, so desktop clients pick up the
change on their next sync cycle.

.. code-block:: console

   occ files_external:notify <mount_id>

See :doc:`external_storage/smb` for setup details, including authentication
requirements and how to reduce sync delay.

**B) All other storages — periodic rescan via cron**

For storage types that do not support push notifications, set up a cron job that
rescans the external storage periodically using its mount ID:

.. code-block:: console

   sudo -E -u www-data php occ files_external:scan <mount_id>

See :doc:`../occ_command` for the full ``files_external:scan`` reference. A typical
interval is every 15 minutes; adjust to balance freshness against server load.

Limitation: renames are not detected reliably
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Whenever Nextcloud rescans an external storage — whether via periodic cron

> (option B) or on direct access (option A setting **Once per direct access**) —

Reviewer: Option A is: _"SMB/CIFS storages — use update notifications"_...
